File: Algorithms_Part_2/Ex12Edge.java
/* *****************************************************************************
 *  Name:
 *  Date:
 *  Description:
 **************************************************************************** */

public class Ex12Edge implements Comparable<Ex12Edge> {

    private final int v, w;
    private final double weight;

    public Ex12Edge(int v, int w, double weight) {
        this.v = v;
        this.w = w;
        this.weight = weight;
    }

    public int either() {
        return v;
    }

    public int other(int vertex) {
        return vertex == this.v ? this.w : this.v;
    }

    public int compareTo(Ex12Edge that) {
        return Double.compare(this.weight, that.weight);
    }

    public double weight() {
        return weight;
    }

    public String toString() {
        return v + " <-- " + weight + " --> " + w;
    }

    public static void main(String[] args) {

    }
}

File: PythonLearnig/Algorithms_Part_2/Ex12Edge.py
class Edge:

    def __init__(self, v, w, weight):
        self.v = v
        self.w = w
        self.weight = weight

    def either(self):
        return self.v

    def other(self, v):
        if v == self.v:
            return self.w
        elif v == self.w:
            return self.v
        else:
            raise ValueError

    def get_weight(self):
        return self.weight

    def compare_to(self, other):
        return self.get_weight() - other.get_weight()

    def __str__(self):
        return f'{self.v} <-- {self.weight} --> {self.w}'


if __name__ == '__main__':
    edge = Edge(2, 7, 0.5)
    print(edge)
